core/modules/catalog/store/product/actions.ts

```diff
@@ -640,7 +641,12 @@ const actions: ActionTree<ProductState, RootState> = {
       throw new Error(`Product query returned empty result product status = ${product.status}`)
     }
     if (product.visibility === 1) { // not visible individually (https://magento.stackexchange.com/questions/171584/magento-2-table-name-for-product-visibility)
-      throw new Error(`Product query returned empty result product visibility = ${product.visibility}`)
+      if (config.products.preventConfigurableChildrenDirectAccess) {
+        const parentProduct = await dispatch('findConfigurableParent', { product })
+        checkParentRedirection(product, parentProduct)
+      } else {
+        throw new Error(`Product query returned empty result product visibility = ${product.visibility}`)
+      }
```

Comment:
The redirect itself seems to work fine. However, I'm thinking whether it would be good to decouple this setting from the visibility configuration.

Having them uncoupled would allow showing a child product on category page, but the link would take you to the parent product.

But this is just my thought. I'm not sure how may shops would actually find this useful.

Reply:
This action is called only on PDP. So with that you can still show simple product on category page and after going on PDP you will be redirected.

Reply:
I also coupled it with visibility because we should check this only when its needed. Right now we are making request for every simple product. With this change it will be done only when you want to get on PDP where simple product shouldn't be visible